File: Practica 2 aux/Composicion y Agregacion/ejercicio 3/Python/Parte.py
class Parte:
    def __init__(self, nombre, peso):
        self.nombre = nombre
        self.peso = peso
    def mostrar(self):
        print("Parte:", self.nombre)
        print("Peso:", self.peso, "kg")
class Avion:
    def __init__(self, modelo, fabricante):
        self.modelo = modelo
        self.fabricante = fabricante
        self.partes = []
    def agrega(self, parte):
        self.partes.append(parte)
    def mostrar(self):
        print("Avion:", self.modelo, "-", self.fabricante)
        for p in self.partes:
            p.mostrar()
a = Avion("Airbus A320", "Airbus")
a.agrega(Parte("Turbina", 2200))
a.agrega(Parte("Tren de aterrizaje", 1300))
a.mostrar()

File: Practica 2 aux/Composicion y Agregacion/ejercicio 3/Java/Parte.java
public class Parte {
    private String nombre;
    private double peso;

    public Parte(String nombre, double peso) {
        this.nombre = nombre;
        this.peso = peso;
    }

    public void mostrar() {
        System.out.println("Parte: " + nombre + ", Peso: " + peso + " kg");
    }
}
